src/transformers/api-history.ts

```diff
@@ -0,0 +1,318 @@
+  return typeof value === 'object' && value !== null && !Array.isArray(value);
+}
+
+function isGithubArtifactsListResponse(
+  value: unknown
+): value is GithubArtifactsListResponse {
+  return isObject(value) && 'artifacts' in value;
+}
+
+function isPrReleaseArtifact(value: unknown): value is PrReleaseArtifact {
+  return isObject(value) && 'data' in value && 'endCursor' in value;
+}
+
+function isApiHistory(value: unknown): value is ApiHistory {
+  return (
+    isObject(value) &&
+    ('added' in value || 'deprecated' in value || 'changes' in value)
+  );
+}
+
+export default function attacher() {
+  return transformer;
+}
+
+function matchApiHistoryCodeBlock(node: Node): node is Code {
+  return (
+    isCode(node) &&
+    node.lang?.toLowerCase() === 'yaml' &&
+    node.meta?.toLowerCase() === 'history'
+  );
+}
+
+let _allElectronVersions: SemVer[];
+
+async function getAllElectronVersions(): Promise<SemVer[]> {
+  if (_allElectronVersions) {
+    return _allElectronVersions;
+  }
+
+  const { versions } = await ElectronVersions.create(undefined, {
+    ignoreCache: true,
+  });
+
+  if (versions == null) throw new Error('No Electron versions found.');
+
+  _allElectronVersions = versions;
+  return _allElectronVersions;
+}
+
+let _allPrReleaseVersions: PrReleaseVersionsContainer;
+
+async function getAllPrReleaseVersions(): Promise<PrReleaseVersionsContainer> {
+  if (_allPrReleaseVersions) {
+    return _allPrReleaseVersions;
+  }
+
+  // TODO: Remove this
+  if (!process.env.GH_TOKEN) {
+    const versions: PrReleaseVersionsContainer = {
+      35658: {
+        release: 'v30.0.0-nightly.20231214',
+        backports: ['v29.0.0-alpha.9'],
+      },
+      41391: {
+        release: 'v31.0.0-alpha.1',
+        backports: [] as string[],
+      },
+      42086: {
+        release: 'v32.0.0-nightly.20240531',
+        backports: ['v31.0.0'],
+      },
+    };
+
+    _allPrReleaseVersions = versions;
+    return _allPrReleaseVersions;
+  }
+
+  if (!process.env.GH_TOKEN) {
+    logger.warn(
+      'No GitHub token found, skipping fetching PR release versions.'
+    );
+    _allPrReleaseVersions = {};
+    return _allPrReleaseVersions;
+  }
+
+  const fetchOptions = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/vnd.github+json',
+      'X-GitHub-Api-Version': '2022-11-28',
+      Authorization: `Bearer ${process.env.GH_TOKEN}`,
+    },
+  };
+
+  const artifactsListResponse = await fetch(
+    GH_ACTIONS_ARTIFACTS_URL,
+    fetchOptions
+  );
+  const artifactsListResponseJson = await artifactsListResponse.json();
+
+  if (!isGithubArtifactsListResponse(artifactsListResponseJson))
+    throw new Error('Invalid GitHub artifacts list response.');
+
+  const latestArtifact = artifactsListResponseJson.artifacts
+    .filter(({ name }) => name === 'resolved-pr-versions')
+    .sort((a, b) => b.id - a.id)[0];
+
+  if (!latestArtifact)
+    throw new Error('No resolved-pr-versions artifact found.');
```

Comment:
```suggestion
  if (!latestArtifact) {
    throw new Error('No resolved-pr-versions artifact found.');
  }
```

Let's avoid this style, it's error-prone.